compiler/codegen/symbol.cpp

```diff
@@ -1104,145 +1104,186 @@ std::string ArgSymbol::getPythonDefaultValue() {
+    res += " = chpl_make_external_array(sizeof(" + typeStrCDefs + "), len(";
+    res += strname + "))\n";
+
+    // Copy the contents over.
+    // E.g. for i in range(len(foo)):
+    //         (<element type*>chpl_foo.elts)[i] = foo[i]
+    res += "\tfor i in range(len(" + strname + ")):\n";
+    res += "\t\t(<" + typeStrCDefs + "*>chpl_" + strname + ".elts)[i] = ";
+    res += strname + "[i]\n";
 
     return res;
-  } else if (t->getValType() == dtExternalArray) {
-    // Handle arrays
-    if (Symbol* eltType = exportedArrayElementType[this]) {
-      // The element type will be recorded in the exportedArrayElementType map
-      // if this arg's type was originally a Chapel array instead of explicitly
-      // an external array.  If we have the element type, that means we need to
-      // do a translation in the python wrapper.
-      std::string typeStr = getPythonTypeName(eltType->type, PYTHON_PYX);
-      std::string typeStrCDefs = getPythonTypeName(eltType->type, C_PYX);
-
-      // Create the memory needed to store the contents of what was passed to us
-      // E.g. cdef chpl_external_array chpl_foo =
-      //          chpl_make_external_array(sizeof(element type), len(foo))
-      std::string res = "\tcdef chpl_external_array chpl_" + strname;
-      res += " = chpl_make_external_array(sizeof(" + typeStrCDefs + "), len(";
-      res += strname + "))\n";
-
-      // Copy the contents over.
-      // E.g. for i in range(len(foo)):
-      //         (<element type*>chpl_foo.elts)[i] = foo[i]
-      res += "\tfor i in range(len(" + strname + ")):\n";
-      res += "\t\t(<" + typeStrCDefs + "*>chpl_" + strname + ".elts)[i] = ";
-      res += strname + "[i]\n";
-
-      return res;
-    }
-  } else if (t->symbol->hasFlag(FLAG_REF) &&
-             t->getValType() == dtOpaqueArray) {
-    // Opaque arrays have a Python representation that stores the C contents in
-    // a field named val
-    std::string res = "\tchpl_" + strname + " = &" + strname + ".val\n";
-    return res;
+  }
+  return "";
+}
 
+//
+// Handle opaque arrays. Opaque arrays have a Python representation that
+// stores the C contents in a field named 'val'.
+//
+static std::string pythonArgToOpaqueArray(ArgSymbol* as) {
+  std::string strname = as->cname;
+  std::string res = "\tchpl_" + strname + " = &" + strname + ".val\n";
+  return res;
+}
+
+//
+// Pass a pointer to a 'numpy.ndarray' buffer or a 'ctypes._Pointer' into
+// Chapel code. In both cases we already know the size of underlying
+// elements.
+//
+static std::string pythonArgToChapelArrayOrPtr(ArgSymbol* as) {
+  Type* t = getArgSymbolCodegenType(as);
+  std::string strname = as->cname;
+
+  // Lydia TODO 12/04/18: Might be good to use a template where we can
+  // replace all instances of a placeholder with the argument name instead of
+  // of writing all of the code out in this chunky, unclear fashion.  Might
+  // be worth considering doing for the else branch above this as well
```

Comment:
This comment needs to be updated, it mentions an else branch that no longer exists due to the code reshuffling.  I think it should point to the other functions instead.